modules/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureBlobContainer.java

```diff
@@ -69,9 +71,12 @@ private InputStream openInputStream(OperationPurpose purpose, String blobName, l
         } catch (Exception e) {
             Throwable rootCause = Throwables.getRootCause(e);
             if (rootCause instanceof BlobStorageException blobStorageException) {
-                if (blobStorageException.getStatusCode() == 404) {
+                if (blobStorageException.getStatusCode() == RestStatus.NOT_FOUND.getStatus()) {
                     throw new NoSuchFileException("Blob [" + blobKey + "] not found");
                 }
+                if (length != null && blobStorageException.getStatusCode() == RestStatus.REQUESTED_RANGE_NOT_SATISFIED.getStatus()) {
```

Comment:
This looks in line with Azure [documentation](https://learn.microsoft.com/en-us/rest/api/storageservices/specifying-the-range-header-for-blob-service-operations#format-1-bytesstartbyte-) which talks about 416 error code.